src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -172,11 +172,21 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
                                # pylint: disable=line-too-long
                                "More details are available at https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki/Username-Password-Authentication")
         raise CLIError(err)
-    except requests.exceptions.SSLError as err:
-        from azure.cli.core.util import SSLERROR_TEMPLATE
-        raise CLIError(SSLERROR_TEMPLATE.format(str(err)))
+    except requests.exceptions.SSLError:
+        from azure.cli.core.azclierror import UnclassifiedUserFault
+        raise UnclassifiedUserFault(error_msg='Certificate verification failed. '
+                                              'This typically happens when using Azure CLI behind a proxy '
+                                              'that intercepts traffic with a self-signed certificate. ',
+                                    recommendation='Please add this certificate to the trusted CA bundle. '
+                                                   'More info: https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively#work-behind-a-proxy.')  # pylint: disable=line-too-long
     except requests.exceptions.ConnectionError as err:
-        raise CLIError('Please ensure you have network connection. Error detail: ' + str(err))
+        from azure.cli.core.azclierror import UnclassifiedUserFault
+        raise UnclassifiedUserFault(error_msg='Connection failure. Error detail: ' + str(err),
+                                    recommendation='Please ensure you have network connection.')
+    except requests.exceptions.InvalidURL:
+        from azure.cli.core.azclierror import UnclassifiedUserFault
+        raise UnclassifiedUserFault(error_msg='Invalid url when login',
+                                    recommendation='Please make sure the cloud is registered with valid url')
```

Comment:
Please use the right punctuation. 

```suggestion
        raise UnclassifiedUserFault(error_msg='Invalid URL.',
                                    recommendation='Please make sure the cloud is registered with a valid Active Directory URL.')
```